src/Idris/SetOptions.idr

```diff
@@ -43,13 +44,15 @@ candidateDirs dname pkg bounds
                             (stringToNatOrZ num :: toVersionNum rest)
 
     getVersion : String -> (String, PkgVersion)
-    getVersion str
-        = case span (/= '-') str of
-               (name, ver) => case strM ver of
-                                   StrNil => (name, MkPkgVersion [0])
-                                   StrCons _ ver =>
-                                      (name, MkPkgVersion (toVersionNum ver))
-               _ => (str, MkPkgVersion [0])
+    getVersion str =
+      -- split the dir name into parts concatenated by "-"
+      -- treating the last part as the version number
+      -- and the initial parts as the package name
+      case Lib.unsnoc $ split (== '-') str of
```

Comment:
Am I wrong to say that the whole deal with hyphenated package names is ambiguous right now, as we also support a naming strategy with missing version strings (for backward compatibility).
Then package my-awesome-package is ambiguous. However, if a version may only be formed of dot separated numbers, then we can parse that without ambiguity (assuming my-package-1 is not a valid package name root)

Reply:
Good catch. I'll have a look.

Reply:
That should do it, thanks !

Reply:
Good catch!